Create MIQROOT/vmdb/vendor/assets symlink and allow Apache to follow

jerryk55 opened the pull request:
the vmdb/vendor/assets directory is used to store compiled assets for the UI.
bin/update via the bower command tries to change ownership on some of these
files but while running from mapped source this will fail.
Create a symlink to the local appliance running on the VM under one's MAC
so the assets are stored locally.

In addition add to the Apache configuration so that symbolic links are followed
properly at all levels.

@roliveri @bdunne @Fryguy please review.  Merge when possible.  Thanks.

jerryk55 committed:
Create MIQROOT/vmdb/vendor/assets symlink and allow Apache to follow

the vmdb/vendor/assets directory is used to store compiled assets for the UI.
bin/update via the bower command tries to change ownership on some of these
files but while running from mapped source this will fail.
Create a symlink to the local appliance running on the VM under one's MAC
so the assets are stored locally.

In addition add to the Apache configuration so that symbolic links are followed
properly at all levels.

roliveri commented on MAC/on_appliance/map-source.sh:
```diff
@@ -58,4 +58,19 @@ echo "**** Run: git update-index --assume-unchanged log/.gitkeep on MAC."
+[[ -d $MIQ_DIR/vendor/assets && ! -L $MIQ_DIR/vendor/assets ]] && rm -rf $MIQ_DIR/vendor/assets
+ln -f -s $LOCAL_VENDOR_ASSETS_DIR $MIQ_DIR/vendor/assets
+
+# Allow Apache to follow the symbolic link created above and set selinux accordingly
```
@jerryk55 - I think this block should be moved into the `software-install.sh` script, after the block that puts selinux in permissive mode. That script changes the configuration if the appliance, while this script just maps MIQ source.

jerryk55 replied:
@roliveri done as suggested.  Note that we also have to source in the defines.sh definitions.

jerryk55 committed:
Move Generation of HTTP Config File to software-install.sh

As per review comments, move the generation of the HTTP configuration
file to allow symbolic links from the map-source.sh script to the
software-install.sh script.
